platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -727,16 +723,7 @@ private ResolverResults resolveGraphIfRequired() {
 
         ResolverResults newState;
         if (!domainObjectContext.getModel().hasMutableState()) {
-            if (!workerThreadRegistry.isWorkerThread()) {
-                // Error if we are executing in a user-managed thread.
-                throw new IllegalStateException("The configuration " + identityPath.toString() + " was resolved from a thread not managed by Gradle.");
-            } else {
-                DeprecationLogger.deprecateBehaviour("Resolution of the configuration " + identityPath.toString() + " was attempted from a context different than the project context. Have a look at the documentation to understand why this is a problem and how it can be resolved.")
-                    .willBecomeAnErrorInGradle9()
-                    .withUserManual("viewing_debugging_dependencies", "sub:resolving-unsafe-configuration-resolution-errors")
-                    .nagUser();
-                newState = domainObjectContext.getModel().fromMutableState(p -> resolveExclusivelyIfRequired());
-            }
+            throw new IllegalStateException("Resolution of the configuration " + identityPath.toString() + " was attempted from a context different than the project context. This is not allowed.");
```

Comment:
I think we need more context about what's wrong here, especially since we've collapsed the two cases.

We could include thread name (if that seems meaningful).  Is there some way we can also include a link to the docs still?

We should also use display name here instead of identity path directly.